fst_iterator.go

```diff
@@ -206,6 +219,9 @@ OUTER:
 
 			cmp := bytes.Compare(i.keysStack, i.nextStart)
 			if cmp > 0 {
+				if dfa, ok := i.aut.(*levenshtein.DFA); ok {
```

Comment:
i think an interface check should do better ...
like 
```
	if distAut, ok := i.aut.(vellum.FuzzyAutomaton); ok {
		i.keysDistance = distAut.EditDistance(autCurr)
	}
```
maybe extending the Automaton interface to support the EditDistance method, which will only be implemented by the DFA automaton